sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -201,11 +222,22 @@ public Read withMaxReadTime(Duration maxReadTime) {
       return toBuilder().setMaxReadTime(maxReadTime).build();
     }
 
+    /**
+     * Specifies the maximum acceptable record lateness.
+     * When this limit is exceeded the runner might decide to scale the amount of resources
+     * allocated to the pipeline in order to speed up ingestion.
+     */
+    public Read withAllowedRecordLateness(Duration allowedRecordLateness) {
```

Comment:
AllowedLateness in Beam refers to event time (i.e. how late a records event time could be compared to current watermark). Here it looks like you want to let user to provide some buffer before the source starts counting the backlog. Could you rename it?
Looks like the current implementation essentially disables backlog if this is > 0, right?